bpf/bpf_lxc.c

```diff
@@ -195,86 +195,120 @@ select_ct_map4(struct __ctx_buff *ctx __maybe_unused, int dir __maybe_unused,
 }
 #endif
 
-#define TAIL_CT_LOOKUP4(ID, NAME, DIR, CONDITION, TARGET_ID, TARGET_NAME)	\
-declare_tailcall_if(CONDITION, ID)						\
-int NAME(struct __ctx_buff *ctx)						\
-{										\
-	struct ct_buffer4 ct_buffer = {};					\
-	int l4_off, ret = CTX_ACT_OK;						\
-	struct ipv4_ct_tuple *tuple;						\
-	struct ct_state *ct_state;						\
-	void *data, *data_end;							\
-	struct iphdr *ip4;							\
-	__u32 zero = 0;								\
-	void *map;									\
-										\
-	ct_state = (struct ct_state *)&ct_buffer.ct_state;			\
-	tuple = (struct ipv4_ct_tuple *)&ct_buffer.tuple;			\
+#if defined ENABLE_IPV4 || defined ENABLE_IPV6
+static __always_inline int drop_for_direction(struct __ctx_buff *ctx,
+					      enum ct_dir dir, __u32 reason)
+{
+	__u32 dst = 0;
+	__u32 dst_id = 0;
+	enum metric_dir m_dir = METRIC_EGRESS;
+	__u32 src_label = 0;
+
+	switch (dir) {
+	case CT_EGRESS:
+		dst_id = 0;
+		dst = 0;
+		src_label = SECLABEL;
+		m_dir = METRIC_EGRESS;
+		break;
+	case CT_INGRESS:
+		dst = SECLABEL;
+		dst_id = LXC_ID;
+		src_label = ctx_load_meta(ctx, CB_SRC_LABEL);
+		m_dir = METRIC_INGRESS;
+		break;
+	/* ingress/egress only for now */
+	default:
+		return send_drop_notify(ctx, src_label, dst, dst_id, reason,
+					CTX_ACT_DROP, m_dir);
+	}
+
+	return send_drop_notify(ctx, src_label, dst, dst_id, reason,
+				CTX_ACT_DROP, m_dir);
+}
+#endif
+
+#define TAIL_CT_LOOKUP4(ID, NAME, DIR, CONDITION, TARGET_ID, TARGET_NAME)     \
+	declare_tailcall_if(CONDITION, ID) int NAME(struct __ctx_buff *ctx)   \
```

Comment:
Let's keep this how it was. It's causing unnecessary code churn.

Reply:
Yeah - autoformatter did this, apologies. Reverted, cleaner diff now.